apex/contrib/csrc/multihead_attn/encdec_multihead_attn_cuda.cu

```diff
@@ -321,9 +318,16 @@ std::vector<torch::Tensor> bwd_cuda(
   char b_layout_t{'t'}; 
   
   //TORCH_CUDABLAS_CHECK(cublasSetMathMode(handle, CUBLAS_TENSOR_OP_MATH));
-  #if USE_GEMM_FLAGS_FP16_ALT_IMPL
-    flags = at::BackwardPassGuard::is_backward_pass() ? rocblas_gemm_flags_fp16_alt_impl : 0;
-  #endif 
+  #ifdef __HIP_PLATFORM_HCC__
+    #define PYTORCH_ROCBLAS_VERSION_DECIMAL (ROCBLAS_VERSION_MAJOR * 100 + ROCBLAS_VERSION_MINOR)
+    #define USE_GEMM_FLAGS_FP16_ALT_IMPL (PYTORCH_ROCBLAS_VERSION_DECIMAL >= 242)
+    #if USE_GEMM_FLAGS_FP16_ALT_IMPL
+      #ifdef ROCM_BACKWARD_PASS_GUARD
+        flags = at::BackwardPassGuard::is_backward_pass() ? rocblas_gemm_flags_fp16_alt_impl : 0;
```

Comment:
This is inside 3 layers of #if conditions, where's the fallback?

Reply:
The variable (default value = 0) defined [here](https://github.com/ROCmSoftwarePlatform/apex/blob/dev/hubertlu/rocblas_backward_compatible/apex/contrib/csrc/multihead_attn/strided_batched_gemm.h#L27) is used in the MHA CUDA files.